factory/django.py

```diff
@@ -142,6 +143,10 @@ def _get_or_create(cls, model_class, *args, **kwargs):
         try:
             instance, _created = manager.get_or_create(*args, **key_fields)
         except IntegrityError as e:
+
+            if cls._original_params is None:
+                raise e
+
```

Comment:
Instead of setting the default value of `_original_params` to `None`, why not set it to `{}`, remove this `if` and let the rest of the code take hold?

```suggestion
```